atomic_reactor/utils/resource_manager.py

```diff
@@ -6,193 +6,308 @@
 of the BSD license. See the LICENSE file for details.
 """
 
-import os
-import logging
-import json
-import paramiko
 import backoff
+import json
 import jsonschema
-from typing import List, Any, Dict, Tuple
+import logging
+import paramiko
+import random
+import os
+from contextlib import contextmanager
 from datetime import datetime
 from shlex import quote
+from typing import List
+
+RESOURCE_LOCK_SCHEMA = 'schemas/resource-lock-content.json'
+SSH_COMMAND_TIMEOUT = 45
+TIMEOUT_FAIL_BUILD = 3000
+RELATIVE_PATH = 'resource_manager'
+RESOURCE_INFO_JSON = 'info.json'
+RETRY_ON_SSH_EXCEPTIONS = (paramiko.ssh_exception.NoValidConnectionsError,
+                           paramiko.ssh_exception.SSHException)
+BACKOFF_FACTOR = 5
+MAX_RETRIES = 3
 
-PARAMIKO_MAX_RETRIES = 3
-PARAMIKO_BACKOFF_FACTOR = 1
+logger = logging.getLogger('resource-manager')
+logger.setLevel(logging.ERROR)
 
-ROOT_PATH = 'resource_manager/'
 
-RESOURCE_LOCK_SCHEMA = 'schemas/resource-lock-content.json'
+class ResourceManagerError(RuntimeError):
+    """ Base module exception """
+
+
+class LockError(ResourceManagerError):
+    """ Attempt for resource locking returned non-zero exitcode """
+
 
-logger = logging.getLogger(__name__)
+class LockRetry(ResourceManagerError):
+    """ Retry locking, no resources were available """
 
 
-class Host:
-    '''
-    Remote host abstract class
-    '''
-    def __init__(self, username: str, hostname: str):
-        self.username = username
-        self.hostname = hostname
-        self.slots: List[Dict[str, Any]] = []
-        self.free_slots: List[int] = []
+class RemoteHost:
 
-    def _read_info_data(self):
-        cmd = f"cat {ROOT_PATH}info.json"
-        data = self._ssh_run_remote_command(cmd)
-        return data
+    def __init__(self, *, hostname: str, username: str, ssh_keyfile: str):
+        """
+        :param hostname: str, remote hostname for ssh connection
+        :param username: str, username for ssh connection
+        :param ssh_keyfile: str, filepath to ssh private key
+        """
+        self._hostname = hostname
+        self._username = username
+        self._ssh_keyfile = ssh_keyfile
 
-    def _read_slot_data(self, slot: int):
-        cmd = f"cat {ROOT_PATH}slot_{slot}.json"
-        data = self._ssh_run_remote_command(cmd)
-        return data
+    @property
+    def hostname(self):
+        return self._hostname
+
+    @property
+    def username(self):
+        return self._username
+
+    @property
+    def ssh_keyfile(self):
+        return self._ssh_keyfile
 
     def _validate_resource_lock(self, payload: dict):
+        """ Validate resource lock json file against schema """
         with open(RESOURCE_LOCK_SCHEMA, 'r') as f:
             schema_data = f.read()
```

Comment:
Will this work after installation? `RESOURCE_LOCK_SCHEMA` is a relative path. We should use `util.validate_with_schema` for this.